C-Plus-Plus/dp/K-Tree.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+problem statement original link: https://codeforces.com/problemset/problem/431/C
+why are we using dynamic programming:
+At every node we are having k different choises to include in our path.
+So whenever we have to deal all the possible choises and pick the best out of them
+then we can use recusrsion . but due to overlapping subproblems it wont be an efficient
+approach.To optimize recusive solution we use dynamic programming.
+
+**dcaught variable says whether there is atleast one edge with weight atleast d in the current path 
+(dp state)
+dp[i][j] means the number of paths from root with sum of edges is i and truth value
+of :there is atleast one edge with weight atleast d is j
+so basically our answer is dp[n][dcaught] (according to dp state).
+*/
```

Comment:
Try to use a more standard language. Check how to write technical documentation. We don't want you to be skilled as a native English speaker, but focus on conveying the problem idea in a minimal way